examples/seismic/tutorials/05_dask.ipynb

```diff
@@ -0,0 +1,496 @@
+    "\n",
+    "This tutorial uses the same synthetic datasets and model setup as the previous two tutorals, so check back if you get lost on parts of the code specific to Devito or SciPy.optimize. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## What is Dask?\n",
+    "\n",
+    "> [Dask](https://dask.pydata.org/en/latest/#dask) is a flexible parallel computing library for analytic computing.\n",
+    ">\n",
+    "> Dask is composed of two components:\n",
+    ">\n",
+    "> * Dynamic task scheduling optimized for computation...\n",
+    "> * “Big Data” collections like parallel arrays, dataframes, and lists that extend common interfaces like NumPy, Pandas, or Python iterators to larger-than-memory or distributed environments. These parallel collections run on top of the dynamic task schedulers.\n",
+    ">\n",
+    "> Dask emphasizes the following virtues:\n",
+    "> \n",
+    "> * Familiar: Provides parallelized NumPy array and Pandas DataFrame objects\n",
+    "> * Flexible: Provides a task scheduling interface for more custom workloads and integration with other projects.\n",
+    "> * Native: Enables distributed computing in Pure Python with access to the PyData stack.\n",
+    "> * Fast: Operates with low overhead, low latency, and minimal serialization necessary for fast numerical algorithms\n",
+    "> * Scales up: Runs resiliently on clusters with 1000s of cores\n",
+    "> * Scales down: Trivial to set up and run on a laptop in a single process\n",
+    "> * Responsive: Designed with interactive computing in mind it provides rapid feedback and diagnostics to aid humans\n",
+    "\n",
+    "**We are going to use it here to parallelise the computation of the functional and gradient as this is the vast bulk of the computational expense of FWI and it is trivially parallel over data shots.**"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setting up (synthetic) data\n",
+    "We are going to set up the same synthetic test case as for the previous tutorial (refer back for details). The code below is slightly re-engineered to make it suitable for using with scipy.optimize.minimize."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "#NBVAL_IGNORE_OUTPUT\n",
+    "from examples.seismic import Model, demo_model\n",
+    "\n",
+    "# Define the grid parameters\n",
+    "def get_grid():\n",
+    "    shape = (101, 101)    # Number of grid point (nx, nz)\n",
+    "    spacing = (10., 10.)  # Grid spacing in m. The domain size is now 1km by 1km\n",
```

Comment:
Maybe make it obvious that domain size is now 101x10=1010m by 1010m

Reply:
>101x101=1010m by 1010m

No it's 1000m by 1000m

Reply:
In that case it is definitely not clear how setting `spacing = (10, 10)` means the domain is now 1000m by 1000m.

Reply:
101 points spaced by 10m, that's 100 cells/interval/... => 1000m

0---10---20---30

4 grrid points, 30m , size is always (n-1)*spacing.

Reply:
^^That should probably be added in a comment here

Reply:
This is pretty standard...take a ruler; it has n spaces but n+1 ticks (grid points in this case).